### PR TITLE
Fixes #22151: API documentation for compliance by directive

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/22151

### CODE REVIEW

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
the CSV part is here :
https://github.com/Normation/rudder/pull/4707/files#diff-1b55ddd0a92eb8832b193d724491a1f21cc872fa2a4118877772f9526729cf6cR7

we can't have 2 same endpoints the describe with the format in another section, maybe @amousset knows a way to doing it

**Comment by VinceMacBuche** [comment]
I think it should be the same with the parameter format and a second response: 
 parameters:
    - in: query
      name: format
      schema:
        type: string
      description: format of export
      style: form
      explode: false
      examples:
        oneId:
          summary: CSV format for the export
          value: [ csv ]   # ?format=csv
    - $ref: ../../components/parameters/directive-id.yml
  responses:
    "200":
      description: Success

**Comment by VinceMacBuche** [comment]
Or at least an example showing it

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
api doc tests do not pass

**Comment by ElaadF** [comment]
PR updated with a new commit

File: webapp/sources/api-doc/components/responses/compliance-directive-id-csv.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2023 Normation SAS
+type: string
+format: binary
+example: >-
```
Comment (amousset) [inline webapp/sources/api-doc/components/responses/compliance-directive-id-csv.yml:5]: I don't think this example is correct

File: webapp/sources/api-doc/components/responses/compliance-directive-id.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          - nodes
+        properties:
+          id:
+            type: string
+            format: uuid
+            description: id of the directive
+            example: 9a1773c9-0889-40b6-be89-f6504443ac1b
+          name:
+            type: string
+            description: name of the directive
```
Comment (amousset) [inline webapp/sources/api-doc/components/responses/compliance-directive-id.yml:44]: you should decide to either start description with an uppercase or not and be consistent with it. IIRC this docs has uppercase.

**Comment by ElaadF** [comment]
PR updated with a new commit

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

**Review by amousset — APPROVED** [review]

**Pull request merged**